cmd/mirror-main.go

```diff
@@ -533,8 +533,15 @@ func (mj *mirrorJob) monitorMirrorStatus(cancel context.CancelFunc) (errDuringMi
 				if isErrIgnored(sURLs.Error) {
 					ignoreErr = true
 				} else {
-					errorIf(sURLs.Error.Trace(sURLs.SourceContent.URL.String()),
-						fmt.Sprintf("Failed to copy `%s`.", sURLs.SourceContent.URL.String()))
+					switch sURLs.Error.ToGoError().(type) {
```

Comment:
Do we anticipate more cases to be added later? Can we use something like `if sURLs.Error.ToGoError().(type) == PathInsufficientPermission` ?

Reply:
we may or may not.

Reply:
also since this is a generic `type` inference it can only be written in `switch` otherwise our linter would have complained to convert this to single `if` condition.